sw/cheri/common/asm.hh

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright lowRISC contributors.
+ * Licensed under the Apache License, Version 2.0, see LICENSE for details.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdint>
+
+namespace ASM{
+  struct Ibex{
+    static void nop(void) {
+      __asm__ volatile ("nop");
+    }
+
+    static void wfi(void) {
+      __asm__ volatile ("wfi");
+    }
+
+    static uint32_t mtime_read(void ){
```

Comment:
The formatting looks weird. Also in C++, there's no need use `void` to mean no parameters.

Reply:
I rebased this PR and applied clang-format, should be fixed now.